src/features/auth/sign-up/ui/sign-up-form.tsx

```diff
@@ -0,0 +1,99 @@
+'use client';
+
+import { useState } from 'react';
+import { useTranslations } from 'next-intl';
+import { useRouter } from '@/shared/i18n/navigation';
+import { buildSignUpRules } from '../model/schema';
+import { useSignUp } from '../model/use-sign-up';
+import { mapSignUpError } from '@/shared/api/firebase/map-sign-up-error';
+
+import { Button, Form, Input, Typography } from 'antd';
+
+const { Item } = Form;
+
+type FieldType = {
+  username?: string;
+  email?: string;
+  password?: string;
+  confirmPassword?: string;
+};
+
+export function SignUpForm() {
+  const t = useTranslations('SignUpForm');
+  const [form] = Form.useForm<FieldType>();
+  const rules = buildSignUpRules(form, t);
+  const router = useRouter();
+
+  const { mutate, loading } = useSignUp();
+  const [apiError, setApiError] = useState<string | null>(null);
+
+  const onFinish = async (values: FieldType) => {
+    try {
+      setApiError(null);
+      await mutate({
+        email: values.email!,
+        password: values.password!,
+        name: values.username,
+      });
```

Comment:
What I wrote about earlier

```suggestion
      await apiSignUp({
        email: values.email!,
        password: values.password!,
        name: values.username,
      });
```